pkg/test/framework/resource/cluster.go

```diff
@@ -30,4 +30,7 @@ type Cluster interface {
 
 	// IsControlPlaneCluster indicates whether or not a control plane has/should have a control plane deployed.
 	IsControlPlaneCluster() bool
+
+	// SharesControlPlane indicates whether or not a cluster can/should share its control plane
+	SharesControlPlane() bool
```

Comment:
Not sure if this really needs to be marked at the `resource.Cluster` level.  The topology inspection could be pushed into `operator.go` but it'd be a little messier.